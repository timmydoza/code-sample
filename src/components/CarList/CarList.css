.CarList {
  width: 100%;
  border-spacing: 0px;
  border-collapse: collapse;
}

.CarList td {
  border: 1px solid lightgrey;
  padding: 0.4em 0.2em;
}

.CarList th {
  width: 25%;
  text-align: left;
  padding: 0.4em 0.2em;
}

td.noBorder {
  border: none;
}

.filterContainer {
  padding: 1em 0;
  overflow: hidden; /*clearfix*/
}


.filterContainer div:first-child {
  float: left;
}

.filterContainer div:last-child {
  float: right;
}

@media (max-width: 550px) {
  .filterContainer div:nth-child(n) {
    float: none;
    padding-bottom: 1em;
  }
}
