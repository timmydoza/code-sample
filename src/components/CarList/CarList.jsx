import React from 'react';
import CarRow from './CarRow/CarRow';
import styles from './CarList.css';
import SortControl from './SortControl/SortControl';
import SearchField from './SearchField/SearchField';
import PaginationButtons from './PaginationButtons/PaginationButtons';

const CarList = (props) => {

  const carRows = props.cars.map((car) => (
    <CarRow
      {...car}
      clickHandler={ () => props.selectCar(car.key) }
    />
  ));

  return (

    <div>

      <SortControl setSort={props.setSort} />

      <SearchField setSearch={props.setSearch} />

      <table className={styles.CarList}>
        <thead>
          <tr>
            <th>Year</th>
            <th>Make</th>
            <th>Model</th>
            <th>Mileage</th>
          </tr>
        </thead>

        <tbody>
          {carRows}
        </tbody>

      </table>

      <PaginationButtons
        totalPages={props.totalPages}
        currentPage={props.currentPage}
        setPage={props.setPage}
      />

    </div>
  );
}

export default CarList