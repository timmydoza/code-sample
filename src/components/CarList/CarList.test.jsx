import React from 'react';
import renderer from 'react-test-renderer';
import CarList from './CarList';

const testProps = {
  currentPage: 3,
  totalPages: 6,
  setPage: jest.fn(),
  setSort: jest.fn(),
  setSearch: jest.fn(),
  cars: [
    {
      key: 1,
      make: 'Testmake',
      model: 'Testmodel',
      year: 1999,
      mileage: 20000
    },
    {
      key: 2,
      make: 'Testmake2',
      model: 'Testmodel2',
      year: 1998,
      mileage: 20001
    }
  ]
};

describe('the CarList component', () => {
  // let paginationButtons, setPage;

  // beforeEach(() => {
  //   setPage = jest.fn();
  //   paginationButtons = renderer
  //     .create(<PaginationButtons currentPage="4" totalPages="11" setPage={setPage}/>)
  //     .toJSON()
  // });

  it('renders correctly', () => {
    const carList = renderer.create(<CarList {...testProps} />).toJSON();
    expect(carList).toMatchSnapshot();
  });

  it('should show a message if there are no results to show', () => {
    const testData = {...testProps};
    testData.cars = [];
    const carList = renderer.create(<CarList {...testData} />).toJSON();
    expect(carList).toMatchSnapshot();
  });

});