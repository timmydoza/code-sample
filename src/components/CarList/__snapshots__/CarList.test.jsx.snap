// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the CarList component renders correctly 1`] = `
<div>
  <div
    className="filterContainer"
  >
    <div
      className="sortControl"
    >
      <label>
        Sort By:
        <select
          onChange={[Function]}
        >
          <option
            value="year"
          >
            Year - Oldest to Newest
          </option>
          <option
            value="mileage"
          >
            Mileage - Highest to Lowest
          </option>
          <option
            value="created_at"
          >
            Date Listed - Newest to Oldest
          </option>
        </select>
      </label>
    </div>
    <div>
      <span>
        Search: 
      </span>
      <input
        onChange={[Function]}
        placeholder="e.g. '2004 honda'"
        type="text"
        value=""
      />
    </div>
  </div>
  <table
    className="CarList"
  >
    <thead>
      <tr>
        <th>
          Year
        </th>
        <th>
          Make
        </th>
        <th>
          Model
        </th>
        <th>
          Mileage
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="CarRow"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1999
        </td>
        <td>
          Testmake
        </td>
        <td>
          Testmodel
        </td>
        <td>
          20000
        </td>
      </tr>
      <tr
        className="CarRow"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <td>
          1998
        </td>
        <td>
          Testmake2
        </td>
        <td>
          Testmodel2
        </td>
        <td>
          20001
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="paginationButtons"
  >
    <button
      onClick={[Function]}
    >
      &lt;
    </button>
    <span>
      3
       of 
      6
    </span>
    <button
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
</div>
`;

exports[`the CarList component should show a message if there are no results to show 1`] = `
<div>
  <div
    className="filterContainer"
  >
    <div
      className="sortControl"
    >
      <label>
        Sort By:
        <select
          onChange={[Function]}
        >
          <option
            value="year"
          >
            Year - Oldest to Newest
          </option>
          <option
            value="mileage"
          >
            Mileage - Highest to Lowest
          </option>
          <option
            value="created_at"
          >
            Date Listed - Newest to Oldest
          </option>
        </select>
      </label>
    </div>
    <div>
      <span>
        Search: 
      </span>
      <input
        onChange={[Function]}
        placeholder="e.g. '2004 honda'"
        type="text"
        value=""
      />
    </div>
  </div>
  <table
    className="CarList"
  >
    <thead>
      <tr>
        <th>
          Year
        </th>
        <th>
          Make
        </th>
        <th>
          Model
        </th>
        <th>
          Mileage
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="noBorder"
        >
          No results found.
        </td>
      </tr>
    </tbody>
  </table>
  <div
    className="paginationButtons"
  >
    <button
      onClick={[Function]}
    >
      &lt;
    </button>
    <span>
      3
       of 
      6
    </span>
    <button
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
</div>
`;
